extension/src/openvic-extension/utility/UITools.cpp

```diff
@@ -123,6 +128,51 @@ static Array get_events_from_shortcut_key(String const& key) {
 	return events;
 }
 
+Error try_create_shortcut_action_for_button(GUIButton* gui_button, String const& shortcut_key_name, String const& shortcut_hotkey_name = "") {
+	Array event_array = get_events_from_shortcut_key(shortcut_key_name);
+
+	ERR_FAIL_COND_V_MSG(
+		shortcut_key_name.length() != 0 && event_array.size() == 0, ERR_INVALID_PARAMETER,
+		vformat("Unknown shortcut key '%s' for GUI button %s", shortcut_key_name, gui_button->get_name())
+	);
+
+	if (shortcut_key_name.length() == 0) {
+		return OK;
+	}
+
+	InputMap* const im = InputMap::get_singleton();
+	String action_name;
+	if (shortcut_hotkey_name.is_empty()) {
+		action_name = //
+			vformat("button_%s_hotkey", gui_button->get_name().to_lower().replace("button", "").replace("hotkey", ""))
+				.replace("__", "_");
```

Comment:
What about a leftover `_` at either end of the name? Currently `close_button` -> `close_`, rather than `close`. Maybe call `replace("__", "_")` after `vformat`, even as well as inside `vformat` (to avoid any risk of a triple `_` that can't be fixed with one `replace`, unless `replace` can actualy handle that case)

Reply:
> Maybe call replace("__", "_") after vformat

It is.